domains/runtime/auto-id/src/lib.rs

```diff
@@ -0,0 +1,885 @@
+    type WeightInfo = pallet_sudo::weights::SubstrateWeight<Runtime>;
+}
+
+parameter_types! {
+    pub const RelayConfirmationDepth: BlockNumber = 18;
+    pub SelfChainId: ChainId = SelfDomainId::self_domain_id().into();
+}
+
+pub struct OnXDMRewards;
+
+impl sp_messenger::OnXDMRewards<Balance> for OnXDMRewards {
+    fn on_xdm_rewards(rewards: Balance) {
+        BlockFees::note_domain_execution_fee(rewards)
+    }
+}
+
+type MmrHash = <Keccak256 as sp_runtime::traits::Hash>::Output;
+
+pub struct MmrProofVerifier;
+
+impl sp_messenger::MmrProofVerifier<MmrHash, Hash> for MmrProofVerifier {
+    fn verify_proof_and_extract_consensus_state_root(
+        opaque_leaf: EncodableOpaqueLeaf,
+        proof: Proof<MmrHash>,
+    ) -> Option<Hash> {
+        let leaf: MmrLeaf<ConsensusBlockNumber, ConsensusBlockHash> =
+            opaque_leaf.into_opaque_leaf().try_decode()?;
+        let state_root = leaf.state_root();
+        verify_mmr_proof(vec![EncodableOpaqueLeaf::from_leaf(&leaf)], proof.encode())
+            .then_some(state_root)
+    }
+}
+
+pub struct StorageKeys;
+
+impl sp_messenger::StorageKeys for StorageKeys {
+    fn confirmed_domain_block_storage_key(domain_id: DomainId) -> Option<Vec<u8>> {
+        get_storage_key(StorageKeyRequest::ConfirmedDomainBlockStorageKey(domain_id))
+    }
+
+    fn outbox_storage_key(chain_id: ChainId, message_key: MessageKey) -> Option<Vec<u8>> {
+        get_storage_key(StorageKeyRequest::OutboxStorageKey {
+            chain_id,
+            message_key,
+        })
+    }
+
+    fn inbox_responses_storage_key(chain_id: ChainId, message_key: MessageKey) -> Option<Vec<u8>> {
+        get_storage_key(StorageKeyRequest::InboxResponseStorageKey {
+            chain_id,
+            message_key,
+        })
+    }
+}
+
+/// Hold identifier for balances for this runtime.
+#[derive(
+    PartialEq, Eq, Clone, Encode, Decode, TypeInfo, MaxEncodedLen, Ord, PartialOrd, Copy, Debug,
+)]
+pub enum HoldIdentifier {
+    Messenger(MessengerHoldIdentifier),
+}
+
+impl VariantCount for HoldIdentifier {
+    const VARIANT_COUNT: u32 = mem::variant_count::<Self>() as u32;
+}
+
+impl pallet_messenger::HoldIdentifier<Runtime> for HoldIdentifier {
+    fn messenger_channel(dst_chain_id: ChainId, channel_id: ChannelId) -> Self {
+        Self::Messenger(MessengerHoldIdentifier::Channel((dst_chain_id, channel_id)))
+    }
+}
+
+parameter_types! {
+    pub const ChannelReserveFee: Balance = 100 * SSC;
+}
+
+impl pallet_messenger::Config for Runtime {
+    type RuntimeEvent = RuntimeEvent;
+    type SelfChainId = SelfChainId;
+
+    fn get_endpoint_handler(endpoint: &Endpoint) -> Option<Box<dyn EndpointHandlerT<MessageId>>> {
+        if endpoint == &Endpoint::Id(TransporterEndpointId::get()) {
+            Some(Box::new(EndpointHandler(PhantomData::<Runtime>)))
+        } else {
+            None
+        }
+    }
+
+    type Currency = Balances;
+    type ConfirmationDepth = RelayConfirmationDepth;
```

Comment:
Also not related to this PR, this item is not used anymore with our  recent change to XDM.